### ISSUE TITLE
aws-s3: eventBridgeEnabled not working as expected. 

### ISSUE THREAD

**Issue by yatin2410** [aws/aws-cdk#26772]
### Describe the bug

Take this simple example : 

`
const bucket = new s3.Bucket(this, 'MyEventBridgeBucket', {
  eventBridgeEnabled: true,
}); 
`

CDK is supposed to enable only event bridge notification. It is available on UI as "Send notifications to Amazon EventBridge for all events in this bucket" but instead CDK is creating IAM role + Policy, Lambda function which I have not defined anywhere and do not want. 

### Expected Behavior

Expected behaviour should be S3 bucket creation with event bridge notification enabled without extra IAM/Lambda resource creations.

### Current Behavior

It is creating extra resources like IAM role+policy, Lambda function which is not mentioned and required to enable event bridge notification. 

### Reproduction Steps

Create new bucket with below code:

`
const bucket = new s3.Bucket(this, 'MyEventBridgeBucket', {
  eventBridgeEnabled: true,
}); 
`

You will be able to see extra resources in cloud formation.

### CDK CLI Version

4.0

### Framework Version

^2.17.0

### Node.js Version

18

### OS

mac OS 13.4

### Language

Typescript

**Comment by pahud** [comment]
Makes sense. Looks like current implementation requires `NotificationsResourceHandler ` as a custom resource handler.

https://github.com/aws/aws-cdk/blob/72481a319b6e71e848e684d79a203c220a393584/packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource.ts#L116

And this will be created once when when `enableEventBridgeNotification()`
https://github.com/aws/aws-cdk/blob/72481a319b6e71e848e684d79a203c220a393584/packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource.ts#L95-L98

I am not sure if `NotificationsResourceHandler` is still required. If yes, we probably still require this custom resource and relevant iam role for the lambda function.
https://github.com/aws/aws-cdk/blob/72481a319b6e71e848e684d79a203c220a393584/packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource.ts#L20-L34

**Comment by hoangduc67** [comment]
same problem on my situation, on enterprise account can not create many lambda as we want, so I think that if enable event bridge it not really need NotificationsResourceHandler then we should not create this lambda

**Comment by blaxx** [comment]
Same problem here. Would appreciate a fix :)

**Comment by pahud** [comment]
reopening this issue as it's still relevant

**Comment by pahud** [comment]
Hi @hoangduc67 @blaxx 

As I mentioned above, the reason we need the custom resource is per description below:

https://github.com/aws/aws-cdk/blob/72481a319b6e71e848e684d79a203c220a393584/packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource.ts#L20-L34

I am asking a second review from the maintainers and see if we could get rid of it.

**Comment by paulhcsun** [comment]
We're not able to remove the `NotificationsResourceHandler` as that would be a breaking change in existing stacks that are using it. If there comes a day when a Cloudformation L1 for it becomes supported then it would be possible to replace the custom resource but currently that's not the case. 

The IAM role + policy do not seem to be overly permissive or anything to me. I understand that it was not something explicitly defined in your stack but as mentioned in 
https://github.com/aws/aws-cdk/blob/72481a319b6e71e848e684d79a203c220a393584/packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource.ts#L20-L34
this is a necessary permission.

**Comment by paulhcsun** [comment]
Apologies I was mistaken about there not being any CFN L1 resources available to replace the custom resource. It should be possible to implement with the following CFN [S3 EventBridgeConfiguration](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-eventbridgeconfiguration.html) resource. 

This would likely be implemented as a new property and we would add a deprecation notice and warning to anyone using the old property. The implementation should also make it possible and not overly difficult for customers using the custom resource to migrate off of it.

**Comment by yerzhan7** [comment]
See also comment here https://github.com/aws/aws-cdk/commit/d8e602b6c1b4cb8ca7038f4b21a7a7092ea8466d for context on why it was done this way: 

> 1. **Why not simply expose EventBridge Cfn property via S3 BucketProps?**
>    Currently CDK manages `NotificationConfigurations `via CustomResource. If we were to expose that way, then e.g. SNS configuration would override EventBridge config.



For now you can do the following workaround:

```
(bucket.node.defaultChild as CfnBucket).notificationConfiguration = {
  eventBridgeConfiguration: {
    eventBridgeEnabled: true,
  },
};
```

But be aware that if you also add SQS/SNS/Lambda notifications (via native CDK L2 methods) on top of EventBridge then CDK will override it.

**Comment by Nicolas-SB** [comment]
We are facing the same problem. 

The workaround seems to work, and if you manually construct and collect all your EventBridge/SQS/SNS/Lambda Notifications in a list and then set them as your notification config you also prevent overwriting anything.

While this works, it would be a lot nicer if CDK just natively supports setting notifications without additional Lambdas.

**Comment by manmartgarc** [comment]
> Apologies I was mistaken about there not being any CFN L1 resources available to replace the custom resource. It should be possible to implement with the following CFN [S3 EventBridgeConfiguration](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-eventbridgeconfiguration.html) resource.
> 
> This would likely be implemented as a new property and we would add a deprecation notice and warning to anyone using the old property. The implementation should also make it possible and not overly difficult for customers using the custom resource to migrate off of it.

this sounds like the way forward - are there any blockers? It seems like the workaround is not really a workaround given the override which violates even snapshot diffs
